apps/src/dance/ai/DanceAiModal.tsx

```diff
@@ -85,6 +95,31 @@ const DanceAiModal: React.FunctionComponent<DanceAiProps> = ({onClose}) => {
     }
   }, [currentAiModalField]);
 
+  useEffect(() => {
+    if (resultJson) {
+      const blocksSvg = generateBlocksFromResult(
+        Blockly.getMainWorkspace(),
+        resultJson
+      );
+
+      const foregroundTranslations = getTranslationsFromField(
+        blocksSvg[0].getField('EFFECT') as FieldDropdown
+      );
+      const backgroundTranslations = getTranslationsFromField(
+        blocksSvg[1].getField('EFFECT') as FieldDropdown
+      );
+      const paletteTranslations = getTranslationsFromField(
+        blocksSvg[1].getField('PALETTE') as FieldDropdown
+      );
+
+      setTranslations({
+        [FieldKey.FOREGROUND_EFFECT]: foregroundTranslations,
```

Comment:
not necessary, but thoughts on converting this into a map of effect ID -> translated label for easier lookup in `AiExplanationView`?

Reply:
Oh I meant it would still be three separate properties, but the values could be maps instead of lists? So doing something like 
```
getTranslationsMap(dropdown: FieldDropdown): {[id: string]: string} {
  const options = dropdown.getOptions();
  const map = {};
  options.forEach(option => {
    map[option[0]] = stripDoubleQuotes(option[1])
  };
  return map;
}
```
So then using that for all three blocks, you'd have a structure that looks like
```
{
  [FieldKey.FOREGROUND_EFFECT]: {
    "bubbles": "Bubbles",
    "raining_tacos": "Tacos"
    "music_notes": "Music Notes",
    ...
  },
  [FieldKey.BACKGROUND_EFFECT]: {
    ...
  },
  ...
}
```

Reply:
This is way better, updated here. Thanks!

https://github.com/code-dot-org/code-dot-org/pull/54534/commits/4650bb89354e9b7e18bdc3df0edced5977943e13